Xamarin.Essentials/Contacts/Contacts.tizen.cs

```diff
@@ -36,56 +38,64 @@ static async Task<Contact> PlatformPickContactAsync()
-                            var phones = new List<ContactPhone>();
-
-                            var recordName = record.GetChildRecord(TizenContact.Name, 0);
-                            if (recordName != null)
-                            {
-                                var first = recordName.Get<string>(TizenName.First) ?? string.Empty;
-                                var last = recordName.Get<string>(TizenName.Last) ?? string.Empty;
-
-                                name = $"{first} {last}".Trim();
-                            }
-
-                            var emailCount = record.GetChildRecordCount(TizenContact.Email);
-                            for (var i = 0; i < emailCount; i++)
-                            {
-                                var item = record.GetChildRecord(TizenContact.Email, i);
-                                var addr = item.Get<string>(TizenEmail.Address);
-                                var type = (TizenEmail.Types)item.Get<int>(TizenEmail.Type);
-
-                                emails.Add(new ContactEmail(addr, GetContactType(type)));
-                            }
-
-                            var phoneCount = record.GetChildRecordCount(TizenContact.Number);
-                            for (var i = 0; i < phoneCount; i++)
-                            {
-                                var item = record.GetChildRecord(TizenContact.Number, i);
-                                var number = item.Get<string>(TizenNumber.NumberData);
-                                var type = (TizenNumber.Types)item.Get<int>(TizenNumber.Type);
-
-                                phones.Add(new ContactPhone(number, GetContactType(type)));
-                            }
-
-                            contact = new Contact(name, phones, emails, ContactType.Unknown);
-                        }
+                            contact = ToContact(record);
                     }
                 }
-
                 tcs.TrySetResult(contact);
             });
-
             return await tcs.Task;
         }
 
+        static async IAsyncEnumerable<Contact> PlatformGetAllAsync()
+        {
+            await Task.CompletedTask;
+
+            var contactsList = manager.Database.GetAll(TizenContact.Uri, 0, 0);
+            for (var i = 0; i < contactsList?.Count; i++)
+            {
+                yield return ToContact(contactsList.GetCurrentRecord());
+                contactsList.MoveNext();
+            }
+        }
+
+        static Contact ToContact(ContactsRecord contactsRecord)
+        {
+            var name = string.Empty;
+            var record = contactsRecord.GetChildRecord(TizenContact.Name, 0);
+            if (record != null)
+            {
+                var first = record.Get<string>(TizenName.First) ?? string.Empty;
+                var last = record.Get<string>(TizenName.Last) ?? string.Empty;
+                name = $"{first} {last}".Trim();
```

Comment:
Can we maybe do something different?

```
if (first)
  if (last)
    first + last
  else 
    first
else
  if (last)
    last
```

The we don't have to allocate strings and then trim. Not perfect, but .NET needs a way to do this...

Reply:
```csharp
if (!string.IsNullOrWhiteSpace(first))
{
     if (!string.IsNullOrWhiteSpace(last))
         name = first + last;
     else
        name = first;
}
else
{
    if (!string.IsNullOrWhiteSpace(last))
        name = last;
}
```

it's not pretty

Reply:
maybe... @dimonovdd 
```csharp
var isFirstValid = !string.IsNullOrWhiteSpace(first);
var isLastValid = !string.IsNullOrWhiteSpace(last);
if (isFirstValid)
{
      name = isLastValid ? first + last : first;
}
else
{
    if (isLastValid)
        name = last;
}
```

Reply:
We need the space, but maybe we could use string.Join...

Reply:
```csharp
string[] names = { contact.NamePrefix, contact.GivenName, contact.MiddleName, contact.FamilyName, contact.NameSuffix };
string displayName = string.Join(' ', names.Where(a => !string.IsNullOrWhiteSpace(a)));
```